client/components/container/container-toolbar-directive.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @copyright Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ *
+ * @fileoverview ContainerToolbarDirective encapsulates layout and UX
+ * for the Container toolbar.
+ *
+ * @author joemu@google.com (Joe Allan Muharsky)
+ */
+
+goog.provide('p3rf.perfkit.explorer.components.container.ContainerToolbarDirective');
+
+goog.require('p3rf.perfkit.explorer.components.explorer.sidebar.SidebarTabService');
+
+
+goog.scope(function() {
+  const explorer = p3rf.perfkit.explorer;
+
+
+  /**
+   * See module docstring for more information about purpose and usage.
+   *
+   * @return {Object} Directive definition object.
+   * @ngInject
+   */
+  explorer.components.container.ContainerToolbarDirective = function(
+      dashboardService) {
+    return {
+      restrict: 'E',
+      replace: true,
+      scope: {},
+      transclude: false,
+      templateUrl: '/static/components/container/container-toolbar-directive.html',
+      controller: ['$scope', function($scope) {
+        this.dashboardSvc = dashboardService;
+
+        /** @export */
+        this.insertContainerBeforeSelected = function() {
+          var container = this.dashboardSvc.selectedContainer;
+          var index = this.dashboardSvc.current.model.children.indexOf(container);
+          this.dashboardSvc.addContainerAt(index);
+        };
+
+        /** @export */
+        this.insertContainerAfterSelected = function() {
+          var container = this.dashboardSvc.selectedContainer;
```

Comment:
Added a TODO #139 to refactor all container code into a single service, and will be done in r24, when the "client" area gets some work.